contrib_bots/lib/DefineBot.py

```diff
@@ -0,0 +1,79 @@
+# See readme.md for instructions on running this code.
+import json
+import requests
+import logging
+
+class DefineHandler(object):
+    '''
+    This plugin define a word that the user inputs. It
+    looks for messages starting with '@define'.
+    '''
+
+    def __init__(self):
+        self._DEFINITION_API_URL = 'https://owlbot.info/api/v1/dictionary/{}?format=json'
+        self._REQUEST_ERROR_MESSAGE = 'Definition not available.'
+        self._EMPTY_WORD_REQUEST_ERROR_MESSAGE = 'Please enter a word to define.'
+
+    def usage(self):
+        return '''
+            This plugin will allow users to define a word. Users should preface
+            messages with "@define".
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        is_define = original_content.startswith('@define')
+
+        return is_define
+
+    def _handle_definition(self, original_content):
+        # Remove '@define' from the message and extract the rest of the message, the
+        # word to define.
+        split_content = original_content.split(' ')
+        to_define = ' '.join(split_content[1:])
+        to_define_lower = to_define.lower()
+
+        # No word was entered.
+        if not to_define_lower:
+            return self._EMPTY_WORD_REQUEST_ERROR_MESSAGE
+        else:
+            response = '**{}**:\n'.format(to_define)
+
+            # Some error occured with requests.get()
+            try:
+                # Use OwlBot API to fetch definition.
+                api_result = requests.get(self._DEFINITION_API_URL.format(to_define_lower))
+                # Convert API result from string to JSON format.
+                definitions = json.loads(api_result.text)
+
+                # Could not fetch definitions for the given word/.
+                if not definitions:
+                    response += self._REQUEST_ERROR_MESSAGE
+                else: # Definitions available.
+                    # Show definitions line by line.
+                    response += '\n'.join('({}) {}'.format(d['type'], d['defenition']) for d in definitions)
```

Comment:
Formatting could be better here if you used markdown, like bullet points for each word, `type` in italics etc. Also you could make use of the `example` value too.